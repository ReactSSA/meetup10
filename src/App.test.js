import React from 'react';
import { shallow } from 'enzyme';
import { cards, deck, card } from './data';
import Card from './components/card';
import App from './App';

describe('<App />', () => {
  describe('render', () => {
    it('Deve renderizar corretamente', () => {
      const wrapper = shallow(<App />);
      expect(wrapper).toMatchSnapshot();
    });

    it('Deve renderizar corretamente com cartas', () => {
      const wrapper = shallow(<App />);
      wrapper.setState({ cards });
      expect(wrapper).toMatchSnapshot();
    });

    it('Deve renderizar corretamente com aberto', () => {
      const wrapper = shallow(<App />);
      wrapper.setState({ isOpen: true, deck });
      expect(wrapper).toMatchSnapshot();
    });
  });

  describe('behavior', () => {
     const mockFetch = global.fetch;

    afterEach(() => {
      global.fetch = mockFetch;
    });

    it('Deve setar o estate corretamente com as cartas', () => {
      const data = {
        cards,
      };

      global.fetch = jest.fn().mockImplementation(() => Promise.resolve({
        json: () => data,
      }));
      const url = 'https://api.magicthegathering.io/v1/cards?page=28';
      const wrapper = shallow(<App />);
      wrapper.instance().componentDidMount();
      expect(fetch).toHaveBeenCalledWith(url);
    });

    it('Deve toggleOpen quando clicado', () => {
      const wrapper = shallow(<App />);
      const span = wrapper.find('.icon');
      span.simulate('click');
      expect(wrapper.state('isOpen')).toBeTruthy();
      span.simulate('click');
      expect(wrapper.state('isOpen')).toBeFalsy();
    });

    it('Deve adicionar uma carta', () => {
      const wrapper = shallow(<App />);
      const { addCard } = wrapper.instance();
      wrapper.setState({ deck });
      addCard(card);
      expect(wrapper.state('deck')).toEqual([
        ...deck,
        card,
      ]);
    });

    it('Deve remover uma carta', () => {
      const wrapper = shallow(<App />);
      const { removeCard } = wrapper.instance();
      wrapper.setState({ deck });
      removeCard(2);
      expect(wrapper.state('deck')).toEqual([
        {
          name: 'Black Lotus',
          cardId: 1,
        },
        {
          name: 'Swamp',
          cardId: 3,
        },
        {
          name: 'Island',
          cardId: 4,
        }
      ]);
    });
  });
});
