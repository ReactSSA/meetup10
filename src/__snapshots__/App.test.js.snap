// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> render Deve renderizar corretamente 1`] = `
<div
  className="appWrapper"
>
  <div
    className="appHeader"
  >
    MONTE SEU DECK DE COMMANDER!
    <span
      className="icon"
      onClick={[Function]}
    >
      Meu deck: 
      0
    </span>
  </div>
</div>
`;

exports[`<App /> render Deve renderizar corretamente com aberto 1`] = `
<div
  className="appWrapper"
>
  <div
    className="appHeader"
  >
    MONTE SEU DECK DE COMMANDER!
    <span
      className="icon"
      onClick={[Function]}
    >
      Meu deck: 
      4
    </span>
  </div>
  <div
    className="wrapper"
  >
    <div
      className="deck"
      key="0"
    >
       - 
      Black Lotus
       
    </div>
    <div
      className="deck"
      key="1"
    >
       - 
      Forest
       
    </div>
    <div
      className="deck"
      key="2"
    >
       - 
      Swamp
       
    </div>
    <div
      className="deck"
      key="3"
    >
       - 
      Island
       
    </div>
  </div>
</div>
`;

exports[`<App /> render Deve renderizar corretamente com cartas 1`] = `
<div
  className="appWrapper"
>
  <div
    className="appHeader"
  >
    MONTE SEU DECK DE COMMANDER!
    <span
      className="icon"
      onClick={[Function]}
    >
      Meu deck: 
      0
    </span>
  </div>
  <Card
    card={
      Object {
        "artist": "murtinha",
        "flavor": "Card flavor",
        "id": 1,
        "image": "path/imagem.png",
        "manaCost": "0",
        "name": "Black Lotus",
        "rarity": "Rare",
        "setName": "Magic collection",
        "text": "Card text",
        "type": "Artifact",
      }
    }
    key="0"
    onAdd={[Function]}
    onRemove={[Function]}
  />
  <Card
    card={
      Object {
        "artist": "murtinha2",
        "flavor": "Card flavor 2",
        "id": 2,
        "image": "path/imagem2.png",
        "manaCost": "B2G",
        "name": "Bufao",
        "rarity": "Legend",
        "setName": "Magic collection 2",
        "text": "Card text 2",
        "type": "Creature",
      }
    }
    key="1"
    onAdd={[Function]}
    onRemove={[Function]}
  />
  <Card
    card={
      Object {
        "artist": "murtinha3",
        "flavor": "Card flavor 3",
        "id": 3,
        "image": "path/imagem3.png",
        "manaCost": "",
        "name": "Forest",
        "rarity": "Basic",
        "setName": "Magic collection 3",
        "text": "Card text 3",
        "type": "Land",
      }
    }
    key="2"
    onAdd={[Function]}
    onRemove={[Function]}
  />
  <Card
    card={
      Object {
        "artist": "murtinha4",
        "flavor": "Card flavor 4",
        "id": 4,
        "image": "path/imagem4.png",
        "manaCost": "",
        "name": "Swamp",
        "rarity": "Basic",
        "setName": "Magic collection 4",
        "text": "Card text 4",
        "type": "Land",
      }
    }
    key="3"
    onAdd={[Function]}
    onRemove={[Function]}
  />
  <Card
    card={
      Object {
        "artist": "murtinha5",
        "flavor": "Card flavor 5",
        "id": 5,
        "image": "path/imagem5.png",
        "manaCost": "",
        "name": "Island",
        "rarity": "Basic",
        "setName": "Magic collection 5",
        "text": "Card text 5",
        "type": "Land",
      }
    }
    key="4"
    onAdd={[Function]}
    onRemove={[Function]}
  />
</div>
`;
